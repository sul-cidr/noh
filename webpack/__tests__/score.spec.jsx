import React from "react";
import { shallow } from "enzyme";
import phrases from "./__fixtures__/phrases.json";
import {
  Unwrapped as UnwrappedScore,
  determineCurrentPhrase
} from "../components/Score";

describe("<UnwrappedScore>", () => {
  const allToggles = {
    isBeatOn: true,
    isTextOn: true,
    isPercussionOn: true,
    isNohkanOn: true,
    isDanceOn: true,
    isPrevSentenceOn: true,
    isNextSentenceOn: true
  };

  it("renders as expected", () => {
    const component = shallow(
      <UnwrappedScore
        phrases={phrases.phrases}
        currentTime={1193}
        toggles={allToggles}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("renders as expected with unmetered phrases", () => {
    const fixturesPhrases = phrases.phrases.slice();
    fixturesPhrases[0].beat = {
      value: "",
      grid: [
        {
          text: "unmetered",
          voices: [],
          start: 12,
          style: "normal",
          length: 1
        }
      ]
    };
    const component = shallow(
      <UnwrappedScore
        phrases={fixturesPhrases}
        currentTime={1193}
        toggles={allToggles}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("determines the correct phrase", () => {
    const component = shallow(
      <UnwrappedScore
        phrases={phrases.phrases}
        currentTime={1193}
        toggles={allToggles}
      />
    );
    const calculatedPhraseIndex = determineCurrentPhrase(
      component.instance().props
    );
    expect(calculatedPhraseIndex).toBe(0);
    const component2 = shallow(
      <UnwrappedScore
        phrases={phrases.phrases}
        currentTime={1216}
        toggles={allToggles}
      />
    );
    const calculatedPhraseIndex2 = determineCurrentPhrase(
      component2.instance().props
    );
    expect(calculatedPhraseIndex2).toBe(2);
    const component3 = shallow(
      <UnwrappedScore
        phrases={phrases.phrases}
        currentTime={0}
        toggles={allToggles}
      />
    );
    const calculatedPhraseIndex3 = determineCurrentPhrase(
      component3.instance().props
    );
    expect(calculatedPhraseIndex3).toBe(0);
  });

  it("sets state correctly on componentWillReceiveProps", () => {
    const component = shallow(
      <UnwrappedScore
        phrases={phrases.phrases}
        currentTime={1213}
        toggles={allToggles}
      />
    );
    component.instance().componentWillReceiveProps(component.instance().props);
    expect(component.state().previousPhrase.phrase).toBe("1");
    expect(component.state().currentPhrase.phrase).toBe("2");
    expect(component.state().nextPhrase.phrase).toBe("3");
  });

  it("handles null state correctly", () => {
    const component = shallow(
      <UnwrappedScore
        phrases={phrases.phrases}
        currentTime={1213}
        toggles={allToggles}
      />
    );
    component.state().currentPhrase = null;
    component.state().nextPhrase = null;
    component.instance().render();
    expect(component).toMatchSnapshot();
  });
});
