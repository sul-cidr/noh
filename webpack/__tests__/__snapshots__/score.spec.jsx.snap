// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnwrappedScore> handles null state correctly 1`] = `
<div
  className="score"
>
  <div
    className="measure measure--previous"
  >
    <MeasureLabelContainer
      isBeatOn={true}
      isDanceOn={true}
      isNextSentenceOn={true}
      isNohkanOn={true}
      isPercussionOn={true}
      isPrevSentenceOn={true}
      isTextOn={true}
      next={false}
      previous={true}
    />
    <div
      className="measure__grid-container"
    >
      <div
        className="measure__channel"
      />
      <div
        className="measure__channel measure__channel--large"
      >
        <ScoreTextLine
          length={0}
          rangeGrid={Array []}
          textGrid={Array []}
        />
      </div>
      <div
        className="measure__channel"
      >
        <CellPercussion
          length={0}
          text="None"
        />
      </div>
      <div
        className="measure__channel"
      >
        <NohkanLine
          grid={Array []}
          length={0}
        />
      </div>
      <div
        className="measure__channel"
      >
        <DanceLine
          grid={Array []}
          length={0}
        />
      </div>
    </div>
  </div>
   
  <div
    className="measure measure--undefined"
  >
    <MeasureLabelContainer
      isBeatOn={true}
      isDanceOn={true}
      isNextSentenceOn={true}
      isNohkanOn={true}
      isPercussionOn={true}
      isPrevSentenceOn={true}
      isTextOn={true}
      next={false}
      previous={false}
      undefined={true}
    />
    <div
      className="measure__grid-container"
    >
      <div
        className="measure__channel"
      >
        <CellBeat
          beatText="unmetered"
          key="beatNum0"
          length={13}
        />
      </div>
      <div
        className="measure__channel measure__channel--large"
      >
        <ScoreTextLine
          length={13}
          rangeGrid={
            Array [
              Object {
                "length": 13,
                "start": 0,
                "style": "normal",
                "text": "l",
                "voices": Array [],
              },
            ]
          }
          textGrid={
            Array [
              Object {
                "length": 1,
                "start": 0,
                "style": "normal",
                "text": "ma",
                "voices": Array [
                  "jiutai",
                ],
              },
              Object {
                "length": 1,
                "start": 1,
                "style": "normal",
                "text": "ta",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 2,
                "style": "normal",
                "text": "wa",
                "voices": Array [
                  "jiutai",
                ],
              },
              Object {
                "length": 1,
                "start": 3,
                "style": "normal",
                "text": "ga",
                "voices": Array [
                  "shite",
                ],
              },
              Object {
                "length": 1,
                "start": 4,
                "style": "normal",
                "text": "cho",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 5,
                "style": "normal",
                "text": "o",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 6,
                "style": "normal",
                "text": "no",
                "voices": Array [
                  "jiutai",
                ],
              },
              Object {
                "length": 1,
                "start": 7,
                "style": "normal",
                "text": "so",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 8,
                "style": "normal",
                "text": "no",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 9,
                "style": "normal",
                "text": "ha",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 10,
                "style": "normal",
                "text": "ji",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 11,
                "style": "normal",
                "text": "me",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 12,
                "style": "normal",
                "text": "(e)",
                "voices": Array [
                  "shite",
                ],
              },
            ]
          }
        />
      </div>
      <div
        className="measure__channel"
      >
        <CellPercussion
          length={13}
          text="mitsuji"
        />
      </div>
      <div
        className="measure__channel"
      >
        <NohkanLine
          grid={
            Array [
              Object {
                "length": 2,
                "start": 1,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 4,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
              Object {
                "length": 2,
                "start": 5,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
              Object {
                "length": 2,
                "start": 10,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
            ]
          }
          length={13}
        />
      </div>
      <div
        className="measure__channel"
      >
        <DanceLine
          grid={
            Array [
              Object {
                "length": 4,
                "start": 1,
                "style": "normal",
                "text": "9. Kneeling facing front.",
                "voices": Array [],
              },
            ]
          }
          length={13}
        />
      </div>
    </div>
  </div>
   
  <div
    className="measure measure--next"
  >
    <MeasureLabelContainer
      isBeatOn={true}
      isDanceOn={true}
      isNextSentenceOn={true}
      isNohkanOn={true}
      isPercussionOn={true}
      isPrevSentenceOn={true}
      isTextOn={true}
      next={true}
      previous={false}
    />
    <div
      className="measure__grid-container"
    >
      <div
        className="measure__channel"
      >
        <CellBeat
          beatText=""
          key="beatNum0"
          length={1}
        />
        <CellBeat
          beatText="1"
          key="beatNum1"
          length={1}
        />
        <CellBeat
          beatText="2"
          key="beatNum2"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum3"
          length={1}
        />
        <CellBeat
          beatText="3"
          key="beatNum4"
          length={1}
        />
        <CellBeat
          beatText="4"
          key="beatNum5"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum6"
          length={1}
        />
        <CellBeat
          beatText="5"
          key="beatNum7"
          length={1}
        />
        <CellBeat
          beatText="6"
          key="beatNum8"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum9"
          length={1}
        />
        <CellBeat
          beatText="7"
          key="beatNum10"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum11"
          length={1}
        />
        <CellBeat
          beatText="8"
          key="beatNum12"
          length={1}
        />
      </div>
      <div
        className="measure__channel measure__channel--large"
      >
        <ScoreTextLine
          length={13}
          rangeGrid={
            Array [
              Object {
                "length": 6,
                "start": 3,
                "style": "normal",
                "text": "m",
                "voices": Array [],
              },
              Object {
                "length": 3,
                "start": 9,
                "style": "normal",
                "text": "l",
                "voices": Array [],
              },
            ]
          }
          textGrid={
            Array [
              Object {
                "length": 1,
                "start": 3,
                "style": "normal",
                "text": "ni",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 4,
                "style": "normal",
                "text": "n",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 5,
                "style": "normal",
                "text": "no",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 6,
                "style": "normal",
                "text": "o",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 7,
                "style": "normal",
                "text": "ji",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 8,
                "style": "normal",
                "text": "u",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 9,
                "style": "normal",
                "text": "ni",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 10,
                "style": "normal",
                "text": "da",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 11,
                "style": "normal",
                "text": "i",
                "voices": Array [],
              },
            ]
          }
        />
      </div>
      <div
        className="measure__channel"
      >
        <CellPercussion
          length={13}
          text=""
        />
      </div>
      <div
        className="measure__channel"
      >
        <NohkanLine
          grid={
            Array [
              Object {
                "length": 2,
                "start": 3,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 8,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
            ]
          }
          length={13}
        />
      </div>
      <div
        className="measure__channel"
      >
        <DanceLine
          grid={
            Array [
              Object {
                "length": 3,
                "start": 0,
                "style": "normal",
                "text": "and gazes in the distance.",
                "voices": Array [],
              },
              Object {
                "length": 2,
                "start": 6,
                "style": "normal",
                "text": "9. Sits back on his knee.",
                "voices": Array [],
              },
            ]
          }
          length={13}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<UnwrappedScore> renders as expected 1`] = `
<div
  className="score"
>
  <div
    className="measure measure--previous"
  >
    <MeasureLabelContainer
      isBeatOn={true}
      isDanceOn={true}
      isNextSentenceOn={true}
      isNohkanOn={true}
      isPercussionOn={true}
      isPrevSentenceOn={true}
      isTextOn={true}
      next={false}
      previous={true}
    />
    <div
      className="measure__grid-container"
    >
      <div
        className="measure__channel"
      />
      <div
        className="measure__channel measure__channel--large"
      >
        <ScoreTextLine
          length={0}
          rangeGrid={Array []}
          textGrid={Array []}
        />
      </div>
      <div
        className="measure__channel"
      >
        <CellPercussion
          length={0}
          text="None"
        />
      </div>
      <div
        className="measure__channel"
      >
        <NohkanLine
          grid={Array []}
          length={0}
        />
      </div>
      <div
        className="measure__channel"
      >
        <DanceLine
          grid={Array []}
          length={0}
        />
      </div>
    </div>
  </div>
   
  <div
    className="measure measure--undefined"
  >
    <MeasureLabelContainer
      isBeatOn={true}
      isDanceOn={true}
      isNextSentenceOn={true}
      isNohkanOn={true}
      isPercussionOn={true}
      isPrevSentenceOn={true}
      isTextOn={true}
      next={false}
      previous={false}
      undefined={true}
    />
    <div
      className="measure__grid-container"
    >
      <div
        className="measure__channel"
      >
        <CellBeat
          beatText=""
          key="beatNum0"
          length={1}
        />
        <CellBeat
          beatText="1"
          key="beatNum1"
          length={1}
        />
        <CellBeat
          beatText="2"
          key="beatNum2"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum3"
          length={1}
        />
        <CellBeat
          beatText="3"
          key="beatNum4"
          length={1}
        />
        <CellBeat
          beatText="4"
          key="beatNum5"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum6"
          length={1}
        />
        <CellBeat
          beatText="5"
          key="beatNum7"
          length={1}
        />
        <CellBeat
          beatText="6"
          key="beatNum8"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum9"
          length={1}
        />
        <CellBeat
          beatText="7"
          key="beatNum10"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum11"
          length={1}
        />
        <CellBeat
          beatText="8"
          key="beatNum12"
          length={1}
        />
      </div>
      <div
        className="measure__channel measure__channel--large"
      >
        <ScoreTextLine
          length={13}
          rangeGrid={
            Array [
              Object {
                "length": 13,
                "start": 0,
                "style": "normal",
                "text": "l",
                "voices": Array [],
              },
            ]
          }
          textGrid={
            Array [
              Object {
                "length": 1,
                "start": 0,
                "style": "normal",
                "text": "ma",
                "voices": Array [
                  "jiutai",
                ],
              },
              Object {
                "length": 1,
                "start": 1,
                "style": "normal",
                "text": "ta",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 2,
                "style": "normal",
                "text": "wa",
                "voices": Array [
                  "jiutai",
                ],
              },
              Object {
                "length": 1,
                "start": 3,
                "style": "normal",
                "text": "ga",
                "voices": Array [
                  "shite",
                ],
              },
              Object {
                "length": 1,
                "start": 4,
                "style": "normal",
                "text": "cho",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 5,
                "style": "normal",
                "text": "o",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 6,
                "style": "normal",
                "text": "no",
                "voices": Array [
                  "jiutai",
                ],
              },
              Object {
                "length": 1,
                "start": 7,
                "style": "normal",
                "text": "so",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 8,
                "style": "normal",
                "text": "no",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 9,
                "style": "normal",
                "text": "ha",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 10,
                "style": "normal",
                "text": "ji",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 11,
                "style": "normal",
                "text": "me",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 12,
                "style": "normal",
                "text": "(e)",
                "voices": Array [
                  "shite",
                ],
              },
            ]
          }
        />
      </div>
      <div
        className="measure__channel"
      >
        <CellPercussion
          length={13}
          text="mitsuji"
        />
      </div>
      <div
        className="measure__channel"
      >
        <NohkanLine
          grid={
            Array [
              Object {
                "length": 2,
                "start": 1,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 4,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
              Object {
                "length": 2,
                "start": 5,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
              Object {
                "length": 2,
                "start": 10,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
            ]
          }
          length={13}
        />
      </div>
      <div
        className="measure__channel"
      >
        <DanceLine
          grid={
            Array [
              Object {
                "length": 4,
                "start": 1,
                "style": "normal",
                "text": "9. Kneeling facing front.",
                "voices": Array [],
              },
            ]
          }
          length={13}
        />
      </div>
    </div>
  </div>
   
  <div
    className="measure measure--next"
  >
    <MeasureLabelContainer
      isBeatOn={true}
      isDanceOn={true}
      isNextSentenceOn={true}
      isNohkanOn={true}
      isPercussionOn={true}
      isPrevSentenceOn={true}
      isTextOn={true}
      next={true}
      previous={false}
    />
    <div
      className="measure__grid-container"
    >
      <div
        className="measure__channel"
      >
        <CellBeat
          beatText=""
          key="beatNum0"
          length={1}
        />
        <CellBeat
          beatText="1"
          key="beatNum1"
          length={1}
        />
        <CellBeat
          beatText="2"
          key="beatNum2"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum3"
          length={1}
        />
        <CellBeat
          beatText="3"
          key="beatNum4"
          length={1}
        />
        <CellBeat
          beatText="4"
          key="beatNum5"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum6"
          length={1}
        />
        <CellBeat
          beatText="5"
          key="beatNum7"
          length={1}
        />
        <CellBeat
          beatText="6"
          key="beatNum8"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum9"
          length={1}
        />
        <CellBeat
          beatText="7"
          key="beatNum10"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum11"
          length={1}
        />
        <CellBeat
          beatText="8"
          key="beatNum12"
          length={1}
        />
      </div>
      <div
        className="measure__channel measure__channel--large"
      >
        <ScoreTextLine
          length={13}
          rangeGrid={
            Array [
              Object {
                "length": 6,
                "start": 3,
                "style": "normal",
                "text": "m",
                "voices": Array [],
              },
              Object {
                "length": 3,
                "start": 9,
                "style": "normal",
                "text": "l",
                "voices": Array [],
              },
            ]
          }
          textGrid={
            Array [
              Object {
                "length": 1,
                "start": 3,
                "style": "normal",
                "text": "ni",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 4,
                "style": "normal",
                "text": "n",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 5,
                "style": "normal",
                "text": "no",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 6,
                "style": "normal",
                "text": "o",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 7,
                "style": "normal",
                "text": "ji",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 8,
                "style": "normal",
                "text": "u",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 9,
                "style": "normal",
                "text": "ni",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 10,
                "style": "normal",
                "text": "da",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 11,
                "style": "normal",
                "text": "i",
                "voices": Array [],
              },
            ]
          }
        />
      </div>
      <div
        className="measure__channel"
      >
        <CellPercussion
          length={13}
          text=""
        />
      </div>
      <div
        className="measure__channel"
      >
        <NohkanLine
          grid={
            Array [
              Object {
                "length": 2,
                "start": 3,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 8,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
            ]
          }
          length={13}
        />
      </div>
      <div
        className="measure__channel"
      >
        <DanceLine
          grid={
            Array [
              Object {
                "length": 3,
                "start": 0,
                "style": "normal",
                "text": "and gazes in the distance.",
                "voices": Array [],
              },
              Object {
                "length": 2,
                "start": 6,
                "style": "normal",
                "text": "9. Sits back on his knee.",
                "voices": Array [],
              },
            ]
          }
          length={13}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<UnwrappedScore> renders as expected with unmetered phrases 1`] = `
<div
  className="score"
>
  <div
    className="measure measure--previous"
  >
    <MeasureLabelContainer
      isBeatOn={true}
      isDanceOn={true}
      isNextSentenceOn={true}
      isNohkanOn={true}
      isPercussionOn={true}
      isPrevSentenceOn={true}
      isTextOn={true}
      next={false}
      previous={true}
    />
    <div
      className="measure__grid-container"
    >
      <div
        className="measure__channel"
      />
      <div
        className="measure__channel measure__channel--large"
      >
        <ScoreTextLine
          length={0}
          rangeGrid={Array []}
          textGrid={Array []}
        />
      </div>
      <div
        className="measure__channel"
      >
        <CellPercussion
          length={0}
          text="None"
        />
      </div>
      <div
        className="measure__channel"
      >
        <NohkanLine
          grid={Array []}
          length={0}
        />
      </div>
      <div
        className="measure__channel"
      >
        <DanceLine
          grid={Array []}
          length={0}
        />
      </div>
    </div>
  </div>
   
  <div
    className="measure measure--undefined"
  >
    <MeasureLabelContainer
      isBeatOn={true}
      isDanceOn={true}
      isNextSentenceOn={true}
      isNohkanOn={true}
      isPercussionOn={true}
      isPrevSentenceOn={true}
      isTextOn={true}
      next={false}
      previous={false}
      undefined={true}
    />
    <div
      className="measure__grid-container"
    >
      <div
        className="measure__channel"
      >
        <CellBeat
          beatText="unmetered"
          key="beatNum0"
          length={13}
        />
      </div>
      <div
        className="measure__channel measure__channel--large"
      >
        <ScoreTextLine
          length={13}
          rangeGrid={
            Array [
              Object {
                "length": 13,
                "start": 0,
                "style": "normal",
                "text": "l",
                "voices": Array [],
              },
            ]
          }
          textGrid={
            Array [
              Object {
                "length": 1,
                "start": 0,
                "style": "normal",
                "text": "ma",
                "voices": Array [
                  "jiutai",
                ],
              },
              Object {
                "length": 1,
                "start": 1,
                "style": "normal",
                "text": "ta",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 2,
                "style": "normal",
                "text": "wa",
                "voices": Array [
                  "jiutai",
                ],
              },
              Object {
                "length": 1,
                "start": 3,
                "style": "normal",
                "text": "ga",
                "voices": Array [
                  "shite",
                ],
              },
              Object {
                "length": 1,
                "start": 4,
                "style": "normal",
                "text": "cho",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 5,
                "style": "normal",
                "text": "o",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 6,
                "style": "normal",
                "text": "no",
                "voices": Array [
                  "jiutai",
                ],
              },
              Object {
                "length": 1,
                "start": 7,
                "style": "normal",
                "text": "so",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 8,
                "style": "normal",
                "text": "no",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 9,
                "style": "normal",
                "text": "ha",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 10,
                "style": "normal",
                "text": "ji",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 11,
                "style": "normal",
                "text": "me",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 12,
                "style": "normal",
                "text": "(e)",
                "voices": Array [
                  "shite",
                ],
              },
            ]
          }
        />
      </div>
      <div
        className="measure__channel"
      >
        <CellPercussion
          length={13}
          text="mitsuji"
        />
      </div>
      <div
        className="measure__channel"
      >
        <NohkanLine
          grid={
            Array [
              Object {
                "length": 2,
                "start": 1,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 4,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
              Object {
                "length": 2,
                "start": 5,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
              Object {
                "length": 2,
                "start": 10,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
            ]
          }
          length={13}
        />
      </div>
      <div
        className="measure__channel"
      >
        <DanceLine
          grid={
            Array [
              Object {
                "length": 4,
                "start": 1,
                "style": "normal",
                "text": "9. Kneeling facing front.",
                "voices": Array [],
              },
            ]
          }
          length={13}
        />
      </div>
    </div>
  </div>
   
  <div
    className="measure measure--next"
  >
    <MeasureLabelContainer
      isBeatOn={true}
      isDanceOn={true}
      isNextSentenceOn={true}
      isNohkanOn={true}
      isPercussionOn={true}
      isPrevSentenceOn={true}
      isTextOn={true}
      next={true}
      previous={false}
    />
    <div
      className="measure__grid-container"
    >
      <div
        className="measure__channel"
      >
        <CellBeat
          beatText=""
          key="beatNum0"
          length={1}
        />
        <CellBeat
          beatText="1"
          key="beatNum1"
          length={1}
        />
        <CellBeat
          beatText="2"
          key="beatNum2"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum3"
          length={1}
        />
        <CellBeat
          beatText="3"
          key="beatNum4"
          length={1}
        />
        <CellBeat
          beatText="4"
          key="beatNum5"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum6"
          length={1}
        />
        <CellBeat
          beatText="5"
          key="beatNum7"
          length={1}
        />
        <CellBeat
          beatText="6"
          key="beatNum8"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum9"
          length={1}
        />
        <CellBeat
          beatText="7"
          key="beatNum10"
          length={1}
        />
        <CellBeat
          beatText=""
          key="beatNum11"
          length={1}
        />
        <CellBeat
          beatText="8"
          key="beatNum12"
          length={1}
        />
      </div>
      <div
        className="measure__channel measure__channel--large"
      >
        <ScoreTextLine
          length={13}
          rangeGrid={
            Array [
              Object {
                "length": 6,
                "start": 3,
                "style": "normal",
                "text": "m",
                "voices": Array [],
              },
              Object {
                "length": 3,
                "start": 9,
                "style": "normal",
                "text": "l",
                "voices": Array [],
              },
            ]
          }
          textGrid={
            Array [
              Object {
                "length": 1,
                "start": 3,
                "style": "normal",
                "text": "ni",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 4,
                "style": "normal",
                "text": "n",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 5,
                "style": "normal",
                "text": "no",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 6,
                "style": "normal",
                "text": "o",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 7,
                "style": "normal",
                "text": "ji",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 8,
                "style": "normal",
                "text": "u",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 9,
                "style": "normal",
                "text": "ni",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 10,
                "style": "normal",
                "text": "da",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 11,
                "style": "normal",
                "text": "i",
                "voices": Array [],
              },
            ]
          }
        />
      </div>
      <div
        className="measure__channel"
      >
        <CellPercussion
          length={13}
          text=""
        />
      </div>
      <div
        className="measure__channel"
      >
        <NohkanLine
          grid={
            Array [
              Object {
                "length": 2,
                "start": 3,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
              Object {
                "length": 1,
                "start": 8,
                "style": "normal",
                "text": "Naka no takane",
                "voices": Array [],
              },
            ]
          }
          length={13}
        />
      </div>
      <div
        className="measure__channel"
      >
        <DanceLine
          grid={
            Array [
              Object {
                "length": 3,
                "start": 0,
                "style": "normal",
                "text": "and gazes in the distance.",
                "voices": Array [],
              },
              Object {
                "length": 2,
                "start": 6,
                "style": "normal",
                "text": "9. Sits back on his knee.",
                "voices": Array [],
              },
            ]
          }
          length={13}
        />
      </div>
    </div>
  </div>
</div>
`;
